lib/core/services/previous-route.service.ts

```diff
@@ -39,4 +41,16 @@ export class PreviousRouteService {
     public getPreviousUrl(): string {
         return this.previousUrl;
     }
+
+    public goBackToPreviousPage(): void {
+        if (this.previousUrl && this.previousUrl.includes('login') || window.history.length <= 2) {
+            this.goBackToHome();
+        } else {
+            this.location.back();
+        }
+    }
+
+    public goBackToHome(): void {
+        this.router.navigate([{outlets: {overlay: null, primary: ['home']}}]);
```

Comment:
what is 'home' from the perspective of another app?

Reply:
also 'login' route might not be also

Reply:
@davidcanonieto yeah, for other apps outside demo shell there is no "home" route :)

Reply:
as suggested previously, navigation commands can give some details regarding location/overlay
```typescript
        const tree = this.router.parseUrl(this.router.url);
        const primaryGroup = tree.root.children[PRIMARY_OUTLET];
        const overlaySegmentGroup = tree.root.children['overlay'];

        console.log(overlaySegmentGroup.toString()); overlay -> /files/something/view
        console.log(primaryGroup.toString()); original location -> // files
```

Reply:
I agree need some love to make it generic can you please collaborate with David in order to improve it?

Reply:
I think the same problem can be found here 
https://github.com/Alfresco/alfresco-ng2-components/blob/development/lib/core/viewer/components/viewer.component.ts#L491

Reply:
yes is the same. for the login route check redirect we should use  :

```
 protected getLoginRoute(): string {
        return (
            this.appConfigService &&
            this.appConfigService.get<string>(
                AppConfigValues.LOGIN_ROUTE,
                'login'
            )
        );
    }
```

For home, we could define a similar strategy in the appconfig or we could just redirect the user to the '/'

Reply:
Can we just pass the url as the route.data ? @pionnegru @eromano

Reply:
From my perspective, we are just trying to close the overlay and we can avoid checking url for login or do other logic if we are just navigate to primary outlet of the overlay.

Reply:
I think this PR is turning into a mess. All we need is to close a password dialog upon pressing ESC.

Reply:
@DenysVuika why is a mess? I removed the routing from inside do you think is a bad idea?

Reply:
@eromano no, that's a good idea :)

Reply:
with the latest commits the overlay is still there and you cannot navigate documentlist. are there any commits to this?

Reply:
@pionnegru you might be looking at the outdated changes? I checked the code and there's no router there.